### ISSUE TITLE
Adding package from pull request URL

### ISSUE THREAD

**Issue by MilesCranmer** [JuliaLang/Pkg.jl#3975]
Just a small convenience, it would be nice if the following syntax worked

```julia
(myproject) pkg> add https://github.com/JuliaPy/PythonCall.jl/pull/529
     Cloning git-repo `https://github.com/JuliaPy/PythonCall.jl/pull/529`
ERROR: failed to clone from https://github.com/JuliaPy/PythonCall.jl/pull/529, error: GitError(Code:ERROR, Class:HTTP, unexpected http status code: 404)
```

Yes it's just a couple more clicks to find the user's repository & branch and install that instead, but it would be nice if you could just input pull requests directly

**Comment by KristofferC** [comment]
Ref https://github.com/JuliaLang/Pkg.jl/pull/3079

**Comment by MilesCranmer** [comment]
For pull request URLs it's a bit more complex since the pull request URL itself does not contain the necessary info. We'd probably have to send an API query to GitHub to get the right remote and branch associated to the pull request.

**Comment by giordano** [comment]
> For pull request URLs it's a bit more complex since the pull request URL

Not really, you can always access a PR branch using the ref `pull/<PR number>/head`.

**Comment by MilesCranmer** [comment]
Cool, that seems like a good way to have this work.
